AU2/test/plugins/util/custom_plugins/test_CompetencyPlugin.py

```diff
@@ -75,18 +76,127 @@ def test_competency_plugin_event_create(self):
+
+        for i in range(10):
+            assert ASSASSINS_DATABASE.get(p[i + 10] + " identifier") in incos
+
+        assert len(incos) == 15
+
+    @plugin_test
+    def test_attempts_auto_competency(self):
+        """
+        Test that attempts grant auto competency properly, especially with multiple.
+        Checks whether a player with 0 or 1 attempt is inco, and one with 2, 3, or 2 spread over different events is not
+        """
+        # TODO refactor this to use Alexei's MockGame.add_attempts() once that is merged
+        p = some_players(5)
+        game = (MockGame().having_assassins(p)
+                .assassin(p[1]).is_involved_in_event(pluginState={"CompetencyPlugin": {"attempts": [p[1] + " identifier"]}})
+                .assassin(p[2]).is_involved_in_event(
+                    pluginState={"CompetencyPlugin": {"attempts": [p[2] + " identifier"] * 2}})
+                .assassin(p[3]).is_involved_in_event(
+                    pluginState={"CompetencyPlugin": {"attempts": [p[3] + " identifier"] * 3}})
+                .assassin(p[4]).is_involved_in_event(
+                    pluginState={"CompetencyPlugin": {"attempts": [p[4] + " identifier"]}})
+                .assassin(p[4]).is_involved_in_event(
+                    pluginState={"CompetencyPlugin": {"attempts": [p[4] + " identifier"]}}))
+
+        manager = self.get_manager(game, auto_competency=True, initial_competency_period=0)
+        query_date = manager.game_start + datetime.timedelta(days=1, seconds=30)
+
+        incos = manager.get_incos_at(query_date)
+        print(EVENTS_DATABASE.events.values())
+
+        for i in range(3):
+            assert ASSASSINS_DATABASE.get(p[i+2] + " identifier") not in incos
+
+        for i in range(2):
+            assert ASSASSINS_DATABASE.get(p[i] + " identifier") in incos
+
+        assert len(incos) == 2
+
+    @plugin_test
+    def test_gigabolt(self):
+        """
+        Test that only non-police players with no kills and no attempts will be eliminated.
+        The dangerous logic for gigabolt is in CompetencyPlugin.gigabolt_ask (i.e. the default selections),
+        while this only tests gigabolt_answer directly. It is mostly just code copy-pasted from gigabolt_ask,
+        and thus is not really a good test
+        """
+        p = some_players(20)
+        game = (MockGame().having_assassins(p)
+                .assassin(p[19]).is_police()
+                .assassin(p[18]).is_police()
+                .assassin(p[0]).kills(p[1])
+                .assassin(p[2]).kills(p[3])
+                .assassin(p[4]).kills(p[5])
+                .assassin(p[6]).kills(p[7])
+                .assassin(p[15]).is_involved_in_event(
+            pluginState={"CompetencyPlugin": {"attempts": [p[15] + " identifier"]}})
+                .assassin(p[16]).is_involved_in_event(
+            pluginState={"CompetencyPlugin": {"attempts": [p[15] + " identifier"] * 3}}))
```

Comment:
Is this meant to be
```suggestion
                .assassin(p[16]).is_involved_in_event(
            pluginState={"CompetencyPlugin": {"attempts": [p[16] + " identifier"] * 3}}))
```
As is the test actually fails because player 16 hasn't actually made any attempts or kills, so there are only 7 live players after gigabolting!

Reply:
Not sure how I didn't catch that...